specifications/cloud_assembly.md

```diff
@@ -0,0 +1,398 @@
+# Cloud Assembly Specification, Version 1.0
+
+The key words **MUST**, **MUST NOT**, **REQUIRED**, **SHALL**, **SHALL NOT**, **SHOULD**, **SHOULD NOT**,
+**RECOMMENDED**, **MAY**, and **OPTIONAL** in this document are to be interpreted as described in [RFC 2119] when they
+are spelled out in bold, capital letters (as they are shown here).
+
+## Introduction
+A *Cloud Assembly* is a self-contained document container designed to hold the components of *cloud applications*,
+including all the parts that are needed in order to deploy those to a *cloud* provider. This document is the
+specification of the *Cloud Assembly* format as well as requirements imposed on *Cloud Assemblers* and *Cloud Runtimes*.
+
+### Design Goals
+The design goals for the *Cloud Assembly Specification* are the following:
+* The *Cloud Assembly Specification* is extensible.
+* The *Cloud Assembly Specification* is cloud-agnostic.
+* The *Cloud Assembly Specification* is easy to implement and use.
+* The *Cloud Assembly Specification* supports authenticity and integrity guarantees.
+* A *Cloud Assembly* is self-contained, making deployments reproductible.
+
+## Specification
+A *Cloud Assembly* is a ZIP archive that **SHOULD** conform to the [ISO/IEC 21320-1:2015] *Document Container File*
+standard. Use of the `deflate` compression method is **RECOMMENDED** in order to minimize the size of the resulting
+file. *Cloud Assembly* files **SHOULD** use the `.cloud` extension in order to make them easier to recognize by users.
+
+Documents in the archive can be stored with any name and directory structure, however the following entries at the root
+of the archive are reserved for special use:
+* `manifest.json` **MUST** be present and contains the [manifest document](#manifest-document) for the *Cloud Assembly*.
+* `signature.asc`, when present, **MUST** contain the [digital signature](#digital-signature) of the *Cloud Assembly*.
+
+### Manifest Document
+The `manifest.json` file is the entry point of the *Cloud Assembly*. It **MUST** be a valid [JSON] document composed of
+a single `object` that conforms to the following schema:
+
+Key           |Type                 |Required|Description
+--------------|---------------------|:------:|-----------
+`schema`      |`string`             |Required|The schema for the document. **MUST** be `cloud-assembly/1.0`.
+`drops`       |`Map<ID,Drop>`       |Required|A mapping of [*Logical ID*](#logical-id) to [Drop](#drop).
```

Comment:
The name "drop" was not at all obvious to me. My first impression was that it would be a set of logical IDs that would be removed ("dropped") from templates somehow. Why not a more obvious name like "component", "artifact", "deployable"...?

Reply:
Would you have been less fooled if `Drop` had been enclosed in backticks, making it look more like a technical name?

Reply:
No. It's that Elad referred to a pun in a different comment, but I had completely missed the punnability of the word "drop".

I think the better term in that case is "droplet", which I think is more commonly understood as a noun that refers to a tiny sphere of water.

Or a more obvious cloud pun like "Cumulus" or whatever?

Reply:
+1 that "drop" for me here reads like a verb, not a noun. "Droplet" is a lot better, however it's already used by EC2... but maybe we don't care.

Reply:
I don't think the collision with EC2 is so material (a search of the combined terms using a popular search engine was not super conclusive). There's other collisions that could be more problematic (but IANAL & I'm not going to discuss this here :D)